src/target/breakpoints.c

```diff
@@ -293,36 +293,49 @@ static void breakpoint_free(struct target *data_target, struct target *breakpoin
 	}
 
 	if (!breakpoint)
-		return;
+		return ERROR_OK;
```

Comment:
```suggestion
		return ERROR_BREAKPOINT_NOT_FOUND;
```

Reply:
This `if` block is dead code; discussed here: https://github.com/riscv/riscv-openocd/pull/967#discussion_r1408102525